organizeImports deletes empty brackets from @types

_From @davidefavia on May 4, 2018 12:50_

Issue Type: <b>Bug</b>

## Expected Result
```ts
import {} from '@types/googlemaps';
```
## Actual Result
```ts
import from '@types/googlemaps';
```
## Steps to reproduce
1. Add below setting to VSCode:
```json
"[typescript]": {
    "editor.codeActionsOnSave": {
        "source.organizeImports": true
     }
}
```
2. Paste code below into editor:
```ts
import {} from '@types/googlemaps';
```
3. Save file and see empty brackets disappeared.

VS Code version: Code 1.23.0 (7c7da59c2333a1306c41e6e7b68d7f0caa7b3d45, 2018-05-03T15:23:14.634Z)
OS version: Darwin x64 17.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-2400S CPU @ 2.50GHz (4 x 2500)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: unavailable_off<br>native_gpu_memory_buffers: unavailable_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>vpx_decode: unavailable_software<br>webgl: unavailable_off<br>webgl2: unavailable_off|
|Load (avg)|2, 2, 2|
|Memory (System)|12.00GB (1.47GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (27)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|0.1.9
vscode-docker|Pet|0.0.26
code-settings-sync|Sha|2.9.2
jsx|Twe|0.0.1
html-snippets|abu|0.2.1
Handlebars|and|0.3.0
vscode-color|ans|0.4.5
vscode-svgviewer|css|1.4.3
angular2-snippets|dan|1.0.19
mustache|daw|1.1.1
vscode-eslint|dba|1.4.8
vscode-yaml-validation|dja|0.0.1
githistory|don|0.4.0
tslint|eg2|1.0.28
vscode-npm-script|eg2|0.3.4
php-debug|fel|1.12.3
php-intellisense|fel|2.3.1
php-pack|fel|1.0.1
terminal|for|0.0.10
jenkins-declarative-support|jmM|0.1.0
Angular1|joh|0.2.0
Angular2|joh|2.16.1
vscode-filesize|mkx|2.0.0
debugger-for-chrome|msj|4.3.0
polacode|pnp|0.2.2
vscode-icons|rob|7.23.0
ReactSnippets|xab|2.1.0


</details>
<!-- generated by issue reporter -->

_Copied from original issue: Microsoft/vscode#49208_

Hi @mhegazy, unfortunately your suggestion doesn't work. VS Code doesn't strip away any code but Angular CLI fails on AOT compilation:
```
Module not found: Error: Can't resolve '@types/googlemaps' in [path-to-component-file]
```
I used the solution suggested by the answer on [SO](https://stackoverflow.com/a/42733315): it works normally, but I didn't find the way to use it with `organizeImports` flag enabled.

--- a/src/harness/unittests/organizeImports.ts
+++ b/src/harness/unittests/organizeImports.ts
@@ -247,6 +247,15 @@ import D from "lib";
                 },
                 libFile);
 
+            testOrganizeImports("Unused_Empty",
+                {
+                    path: "/test.ts",
+                    content: `
+import { } from "lib";
+`,
+                },
+                libFile);
+
             testOrganizeImports("Unused_false_positive_shorthand_assignment",
             {
                     path: "/test.ts",

--- a/src/services/organizeImports.ts
+++ b/src/services/organizeImports.ts
@@ -198,7 +198,9 @@ namespace ts.OrganizeImports {
             : namedImports[0];
 
         const newNamedImports = sortedImportSpecifiers.length === 0
-            ? undefined
+            ? newDefaultImport
+                ? undefined
+                : createNamedImports(emptyArray)
             : namedImports.length === 0
                 ? createNamedImports(sortedImportSpecifiers)
                 : updateNamedImports(namedImports[0].importClause.namedBindings as NamedImports, sortedImportSpecifiers);

--- a/tests/baselines/reference/organizeImports/Unused_Empty.ts
+++ b/tests/baselines/reference/organizeImports/Unused_Empty.ts
@@ -1,0 +1,7 @@
+// ==ORIGINAL==
+
+import { } from "lib";
+
+// ==ORGANIZED==
+
+import { } from "lib";

